### PR TITLE
CB-10636 Add JSHint for plugins

### PR DESCRIPTION
**Pull request by daserge**
Added Travis badge
Added node_modules to gitignore
Disabled quotmark as it is [deprecated](http://jshint.com/docs/options/#quotmark)
Fixed jshint issues

[Jira issue](https://issues.apache.org/jira/browse/CB-10636)

### CODE REVIEW

**Comment by daserge** [comment]
There is one issue remaining:
https://github.com/apache/cordova-plugin-device/blob/52f0b41d841238659d7c7f32d60d9231718b27ac/src/ubuntu/device.js#L28

`src\ubuntu\device.js: line 27, col 24, 'name' is not defined.`

**Comment by daserge** [comment]
@vladimir-kotikov, could you please take a look?

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+    "node": true
```
Comment (vladimir-kotikov) [inline .jshintrc]: Should't we use `browser: true` instead and add `define`, `module` and `require` to `globals`?

File: .jshintrc [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  , "bitwise": true
+  , "undef": true
+  , "trailing": true
+  , "quotmark": false
+  , "indent": 4
+  , "unused": "vars"
+  , "latedef": "nofunc"
+  , "jasmine": true
```
Comment (vladimir-kotikov) [inline .jshintrc]: Setting `jasmine` here seems excess, as there is only one file with tests. Maybe just add inline `/* jshint jasmine: true*/` to `tests.js`

**Comment by daserge** [comment]
Addressed, thanks for review, @vladimir-kotikov.
BTW, jshint does not highlight the `src\ubuntu\device.js: line 27, col 24, 'name' is not defined.` issue with `browser: true` setting.

**Comment by vladimir-kotikov** [comment]
I think this is because of https://developer.mozilla.org/docs/Web/API/Window/name

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+language: node_js
+sudo: false
+node_js:
+  - "0.10"
```
Comment (nikhilkh) [inline .travis.yml]: There is no point in running jshint for all 3 versions of node.js.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+  - "0.12"
+  - "4.2"
```
Comment (dblotsky) [inline .travis.yml]: We don't need to test on different Node versions, since there is no Node code in plugins.

**Comment by daserge** [comment]
@nikhilkh, @dblotsky, addressed, thanks!

**Pull request merged**